test_conformance/spirv_new/test_op_spec_constant.cpp

```diff
@@ -0,0 +1,182 @@
+Copyright (c) 2016 The Khronos Group Inc. All Rights Reserved.
+
+This code is protected by copyright laws and contains material proprietary to
+the Khronos Group, Inc. This is UNPUBLISHED PROPRIETARY SOURCE CODE that may not
+be disclosed in whole or in part to third parties, and may not be reproduced,
+republished, distributed, transmitted, displayed, broadcast or otherwise
+exploited in any manner without the express prior written permission of Khronos
+Group. The receipt or possession of this code does not convey any rights to
+reproduce, disclose, or distribute its contents, or to manufacture, use, or sell
+anything that it may describe, in whole or in part other than under the terms of
+the Khronos Adopters Agreement or Khronos Conformance Test Source License
+Agreement as executed between Khronos and the recipient.
+******************************************************************/
+
+#include "testBase.h"
+#include "types.hpp"
+
+
+template <typename T>
+int test_spec_constant(cl_device_id deviceID, cl_context context,
+                       cl_command_queue queue, const char *name,
+                       std::vector<T> &inputs, std::vector<T> &references,
+                       T spec_constant_value,
+                       bool (*notEqual)(const T &, const T &) = isNotEqual<T>)
+{
+    if (std::string(name).find("double") != std::string::npos)
+    {
+        if (!is_extension_available(deviceID, "cl_khr_fp64"))
+        {
+            log_info("Extension cl_khr_fp64 not supported; skipping double "
+                     "tests.\n");
+            return TEST_SKIPPED_ITSELF;
+        }
+    }
+    if (std::string(name).find("half") != std::string::npos)
+    {
+        if (!is_extension_available(deviceID, "cl_khr_fp16"))
+        {
+            log_info(
+                "Extension cl_khr_fp16 not supported; skipping half tests.\n");
+            return TEST_SKIPPED_ITSELF;
+        }
+    }
+
+    clProgramWrapper prog;
+    cl_int err = get_program_with_il(prog, deviceID, context, name);
+    SPIRV_CHECK_ERROR(err, "Failed to build program");
+
+    clKernelWrapper kernel = clCreateKernel(prog, "spec_const_kernel", &err);
+    SPIRV_CHECK_ERROR(err, "Failed to create kernel");
+    int num = (int)references.size();
+    size_t bytes = num * sizeof(T);
+    clMemWrapper numbers_buffer =
+        clCreateBuffer(context, CL_MEM_READ_WRITE | CL_MEM_USE_HOST_PTR, bytes,
+                       inputs.data(), &err);
+    SPIRV_CHECK_ERROR(err, "Failed to create numbers_buffer");
+
+    clMemWrapper spec_buffer =
+        clCreateBuffer(context, CL_MEM_USE_HOST_PTR,
+                       sizeof(spec_constant_value), &spec_constant_value, &err);
+    SPIRV_CHECK_ERROR(err, "Failed to create spec_buffer");
+
+    err = clSetKernelArg(kernel, 0, sizeof(clMemWrapper), &numbers_buffer);
+    SPIRV_CHECK_ERROR(err, "Failed to set kernel argument inputs_buffer");
+    err = clSetKernelArg(kernel, 1, sizeof(clMemWrapper), &spec_buffer);
```

Comment:
This `spec_buffer` argument does not seem to be used for anything, in the kernels that I have looked at. Could you explain its purpose, please, or remove it if it has none?

Reply:
Good catch. It looks really not needed.